src/org/oidc/common/MissingRequiredAttribute.java

```diff
@@ -0,0 +1,7 @@
+package org.oidc.common;
+
+public class MissingRequiredAttribute extends Exception {
```

Comment:
Like I have mentioned in previous comments as well - please name custom exception classes ending with Exception. That's the general convention.

Instead of MissingRequiredAttribute --> MissingRequiredAttributeException

Reply:
that seems to be outdated.  pls see new.